### ISSUE TITLE
no-useless-escape not allowing whitespace regex character

### ISSUE THREAD

**Issue by JoshuaHall** [eslint/eslint#11143]
**What rule do you want to change?**
no-useless-escape
**Does this change cause the rule to produce more or fewer warnings?**
Fewer
**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behavior
**Please provide some example code that this change will affect:**

```js
const whitespaceRegEx = new RegExp('^\s+$', 'g');
```

**What does the rule currently do for this code?**
Currently, the rule will warn that \s does not not need to be escaped. \s is a special regex character to select any whitespace character.
**What will the rule do after it's changed?**
The rule would not by default display an error.
**Are you willing to submit a pull request to implement this change?**
I am willing to.

**Comment by platinumazure** [comment]
Hi @JoshuaHall, thanks for the issue.

This is working as intended. You're using the `RegExp` constructor with a string literal, so the string escape rules apply, not the escape rules for regex literals.

The following constructions are equivalent:

* `/^\s+$/g`
* `new RegExp('^\\s+$', 'g')`

Note that you need to double up the backslash in the string.
